docs/docs/api/arg.md

```diff
@@ -4,11 +4,31 @@ title: Arg
 sidebar_label: Arg
 ---
 
-`Arg` allows referring to other arguments in header of a
-[producer](/docs/api/producer) or [view](/docs/api/view).
+```ts
+import { Arg } from "@c11/engine.macro"
+```
+
+## Overview
+
+`Arg` allows referring to other arguments in the header of a
+[producer](/docs/api/producer) or [view](/docs/api/view). It makes it possible
+to treat other keys of the header object as local variables. All of the
+following are valid uses of `Arg`:
+
+```ts
+        const result: producer = ({
+          a1 = '123',
+          a2 = Arg.a1,            // Access previously defined argument
+          a3 = Arg.b1.b2.b3.b4    // Access nested properties of another Arg
```

Comment:
I believe this is the case with Get, Observe, and Update as well. I am documenting this behavior in `concepts/path-composition`, since this behavior is going to span over any (possible) future operators as well. Accessing non-existent state is after all a recommended pattern in Engine. Please let me know if this should be different.